tests/run-make/dep-info/rmake.rs

```diff
@@ -0,0 +1,41 @@
+// This is a simple smoke test for rustc's `--emit dep-info` feature. It prints out
+// information about dependencies in a Makefile-compatible format, as a `.d` file.
+
+// This test first checks that emitting dep-info disables static analysis, preventing
+// compilation of `erroneous.rs` from causing a compilation failure.
+// Then, it checks that compilation using the flag is successful in general, even with
+// empty source files or source files that contain a whitespace character.
+
+// Finally, it removes one dependency and checks that compilation is still successful.
+// See https://github.com/rust-lang/rust/pull/10698
+
+// FIXME try it on windows and freebsd
+// on windows `rustc --dep-info` produces Makefile dependency with
+// windows native paths (e.g. `c:\path\to\libfoo.a`)
+// but msys make seems to fail to recognize such paths, so test fails.
```

Comment:
Remark: in msys2, the path might actually be a unix path... so our host (`uname -a`) is

```bash
$ uname -a
MSYS_NT-10.0-22631 Joe 3.5.3.x86_64 2024-06-03 06:22 UTC x86_64 Msys
```
 
<img width="617" alt="Screenshot 2024-08-11 111003" src="https://github.com/user-attachments/assets/7071190e-fd1f-4812-aa74-3cd75400660f">